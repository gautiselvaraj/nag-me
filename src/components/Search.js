import React from 'react';
import styled from 'styled-components';
import Input from './Input';

const Search = styled.div`
  position: relative;

  input {
    padding-left: 20px;
  }

  &:before {
    color: ${props => props.theme.greyLight};
    content: '\\e986';
    font-family: icons;
    font-size: 0.5rem;
    left: 7px;
    position: absolute;
    top: 7px;
  }
`;

export default ({ handleKeyup }) => (
  <Search>
    <Input
      type="search"
      small
      search
      placeholder="Search"
      onKeyUp={handleKeyup}
    />
  </Search>
);
