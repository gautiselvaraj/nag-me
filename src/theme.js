export default {
  main:           '#214154',
  mainDarken:     '#083148',
  mainLighten:    '#215a7b',
  black:          '#111111',
  white:          '#f6f6f6',
  greyLightest:   '#efefef',
  greyLighter:    '#dddddd',
  greyLight:      '#a5a5a5',
  grey:           '#898989',
  greyDark:       '#777777',
  greyDarker:     '#595959',
  greyDarkest:    '#222',
}
