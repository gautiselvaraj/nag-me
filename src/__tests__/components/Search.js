import React from 'react';
import { mount } from 'enzyme';
import Search from '../../components/Search';
import theme from '../../theme';
import 'jest-styled-components';

describe('<Search />', () => {
  it('should match snapshot and have correct elements & styles', () => {
    let search = mount(<Search theme={theme} />);
    expect(search).toMatchSnapshot();
    expect(search.find('input[placeholder="Search"]').length).toBe(1);
    expect(search).toHaveStyleRule('content', "'\\e986'", {
      modifier: ':before'
    });
  });
});
