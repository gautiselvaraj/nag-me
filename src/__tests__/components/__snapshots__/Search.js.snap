// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Search /> should match snapshot and have correct elements & styles 1`] = `
.c0 {
  position: relative;
}

.c0 input {
  padding-left: 20px;
}

.c0:before {
  content: '\\e986';
  font-family: icons;
  font-size: 0.5rem;
  left: 7px;
  position: absolute;
  top: 7px;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.c2 {
  border: 1px solid #dddddd;
  border-radius: 5px;
  box-sizing: border-box;
  font-size: .85rem;
  height: 1.5rem;
  padding: 0 0.35rem;
  width: 100%;
}

<Component
  theme={
    Object {
      "black": "#111111",
      "grey": "#898989",
      "greyDark": "#777777",
      "greyDarker": "#595959",
      "greyDarkest": "#222",
      "greyLight": "#a5a5a5",
      "greyLighter": "#dddddd",
      "greyLightest": "#efefef",
      "main": "#214154",
      "mainDarken": "#08283B",
      "mainLighten": "#3B5B6E",
      "white": "#f6f6f6",
    }
  }
>
  <styled.div>
    <div
      className="c0"
    >
      <Component
        placeholder="Search"
        small={true}
        theme={
          Object {
            "black": "#111111",
            "grey": "#898989",
            "greyDark": "#777777",
            "greyDarker": "#595959",
            "greyDarkest": "#222",
            "greyLight": "#a5a5a5",
            "greyLighter": "#dddddd",
            "greyLightest": "#efefef",
            "main": "#214154",
            "mainDarken": "#08283B",
            "mainLighten": "#3B5B6E",
            "white": "#f6f6f6",
          }
        }
        type="search"
      >
        <styled.div
          placeholder="Search"
          small={true}
          theme={
            Object {
              "black": "#111111",
              "grey": "#898989",
              "greyDark": "#777777",
              "greyDarker": "#595959",
              "greyDarkest": "#222",
              "greyLight": "#a5a5a5",
              "greyLighter": "#dddddd",
              "greyLightest": "#efefef",
              "main": "#214154",
              "mainDarken": "#08283B",
              "mainLighten": "#3B5B6E",
              "white": "#f6f6f6",
            }
          }
          type="search"
        >
          <div
            className="c1"
            placeholder="Search"
            type="search"
          >
            <styled.input
              placeholder="Search"
              small={true}
              theme={
                Object {
                  "black": "#111111",
                  "grey": "#898989",
                  "greyDark": "#777777",
                  "greyDarker": "#595959",
                  "greyDarkest": "#222",
                  "greyLight": "#a5a5a5",
                  "greyLighter": "#dddddd",
                  "greyLightest": "#efefef",
                  "main": "#214154",
                  "mainDarken": "#08283B",
                  "mainLighten": "#3B5B6E",
                  "white": "#f6f6f6",
                }
              }
              type="search"
            >
              <input
                className="c2"
                placeholder="Search"
                type="search"
              />
            </styled.input>
          </div>
        </styled.div>
      </Component>
    </div>
  </styled.div>
</Component>
`;
